lib/sensors/gyroscope.cpp

```diff
@@ -0,0 +1,131 @@
+    : logger_(logger),
+      i2c_(i2c),
+      channel_(channel),
+      device_address_(device_address)
+{
+}
+
+Gyroscope::~Gyroscope()
+{
+}
+
+const std::optional<std::int16_t> Gyroscope::read(core::Axis axis)
+{
+  switch (axis) {
+    case core::Axis::kX: {
+      const auto gyroscope_x_high_byte = i2c_->readByte(device_address_, kDataXHigh);
+      if (!gyroscope_x_high_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope X-axis high at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto gyroscope_x_low_byte = i2c_->readByte(device_address_, kDataXLow);
+      if (!gyroscope_x_low_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope X-axis low at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto x_axis = ((*gyroscope_x_high_byte << 8) | *gyroscope_x_low_byte);
+      logger_.log(
+        core::LogLevel::kDebug, "Successfully read x-axis from gyroscope at channel %d", channel_);
+      return static_cast<std::uint16_t>(x_axis * 0.00875);
+    }
+    case core::Axis::kY: {
+      const auto gyroscope_y_high_byte = i2c_->readByte(device_address_, kDataYHigh);
+      if (!gyroscope_y_high_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope Y-axis high at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto gyroscope_y_low_byte = i2c_->readByte(device_address_, kDataYLow);
+      if (!gyroscope_y_low_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope Y-axis low at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto y_axis = ((*gyroscope_y_high_byte << 8) | *gyroscope_y_low_byte);
+      logger_.log(
+        core::LogLevel::kDebug, "Successfully read y-axis from gyroscope at channel %d", channel_);
+      return static_cast<std::uint16_t>(y_axis * 0.00875);
+    }
+    case core::Axis::kZ: {
+      const auto gyroscope_z_high_byte = i2c_->readByte(device_address_, kDataZHigh);
+      if (!gyroscope_z_high_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope Z-axis high at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto gyroscope_z_low_byte = i2c_->readByte(device_address_, kDataZLow);
+      if (!gyroscope_z_low_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope Z-axis low at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto z_axis = ((*gyroscope_z_high_byte << 8) | *gyroscope_z_low_byte);
+      logger_.log(
+        core::LogLevel::kDebug, "Successfully read z-axis from gyroscope at channel %d", channel_);
+      return static_cast<std::uint16_t>(z_axis * 0.00875);
+    }
+    default: {
+      logger_.log(core::LogLevel::kFatal, "Gave an invalid axis");
+      return std::nullopt;
+    }
+  }
```

Comment:
i think this should probably be restructured to have a `readAxis` helper function and maybe pass in `kDataXHigh` and `kDataXLow` as parameters, right now this is a lot of almost duplicate code.